.github/workflows/retry-release.yml

```diff
@@ -0,0 +1,79 @@
+name: Retry Release
+on:
+  workflow_dispatch:
+    inputs:
+      release:
+        description: The Grails Release (e.g. 5.0.3)
+        required: true
+      target_branch:
+        description: The Target Branch (e.g. 5.0.x)
+        required: true
+permissions: {}
+jobs:
+  release:
+    permissions:
+      contents: write  #  to create release
+      issues: read  #  to get closed issues
+
+    runs-on: ubuntu-latest
+    env:
+      GIT_USER_NAME: puneetbehl
+      GIT_USER_EMAIL: behlp@unityfoundation.io
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          ref: "v${{ github.event.inputs.release }}"
+          token: ${{ secrets.GH_TOKEN }}
+      - name: Set up JDK
+        uses: actions/setup-java@387ac29b308b003ca37ba93a6cab5eb57c8f5f93 # v4
```

Comment:
Please use `actions/setup-java@v4` instead of a hash because `v4` is updated to the latest hash automatically.